core/src/main/java/org/commonjava/maven/galley/io/SpecialPathManagerImpl.java

```diff
@@ -249,11 +250,17 @@ public SpecialPathInfo getSpecialPathInfo( String path )
         return path == null ? null : getSpecialPathInfo( null, path, SpecialPathConstants.PKG_TYPE_MAVEN );
     }
 
-    public void setCacheProvider( CacheProvider provider )
+    private String getStoragePath( ConcreteResource resource )
     {
-        if ( this.cacheProvider == null )
+        String storagePath = resource.getPath();
+        if ( pathGenerators != null )
         {
-            this.cacheProvider = provider;
+            for ( final PathGenerator generator : pathGenerators )
```

Comment:
Do we have a use case for passing in multiple `PathGenerator` instances? Won't this logic simply take the `storagePath` from the last `PathGenerator` it iterates over? If we don't need multiple PathGenerator instances, we should simplify down to just one and pass that in, I think.

Reply:
I see...they will accumulate. But keep in mind that PathGenerator != StoragePathCalculator (in Indy)...the Indy implementation of PathGenerator uses multiple StoragePathCalculator instances.

Reply:
Yes that's concern to me. But as PathGenerator here is an interface, which we injected from external through Instance<>, I'm not sure if there will be more instances of it in the future. So that's why I used Set and Iterates here. I'm also not sure if it is a good idea to set storagePath base on previous storagePath, but from the interface meaning it seems acting like so.